plugins/maciVoting/hooks/useCanFinalize.tsx

```diff
@@ -1,51 +1,14 @@
 import { useState, useEffect } from "react";
-import { getPoll } from "@maci-protocol/sdk/browser";
-import { PUBLIC_MACI_ADDRESS } from "@/constants";
-import { useEthersSigner } from "./useEthersSigner";
-import { usePublicClient } from "wagmi";
-import { useCoordinator } from "./useCoordinator";
+import { useGetPollData } from "./useGetPollData";
 
 export const useCanFinalize = (pollId?: bigint) => {
   const [canFinalize, setCanFinalize] = useState(false);
-  const { checkIsTallied } = useCoordinator();
 
-  const publicClient = usePublicClient();
-  const signer = useEthersSigner();
+  const { data: { voteEnded, tallied } = {} } = useGetPollData(pollId);
```

Comment:
I refactored `useCanFinalize` to directly use the values returned by `useGetPollData` so all calls to the results, voteEnded, tallied, etc will be made from `useGetPollData` (one source of truth). It is useful for debugging